packages/react-utilities/src/hooks/useTriggerElement.ts

```diff
@@ -0,0 +1,102 @@
+import * as React from 'react';
+
+import { applyTriggerPropsToChildren } from '../utils/applyTriggerPropsToChildren';
+import { getReactCallbackName } from '../utils/getReactCallbackName';
+import { onlyChild } from '../utils/onlyChild';
+import { useEventCallback } from './useEventCallback';
+import { useMergedRefs } from './useMergedRefs';
+import type { ReactCallbackName } from '../utils/getReactCallbackName';
+
+export type UseTriggerElementOptions<TriggerProps> = {
+  /** An actual trigger element or render props function. */
+  children: React.ReactElement | ((props: TriggerProps) => React.ReactNode) | null | undefined;
+
+  /** A ref to trigger element. */
+  ref: React.Ref<unknown> | undefined;
+
+  /** Props that are passed to a parent component and should be forwarded down. */
+  outerProps: React.HTMLProps<unknown>;
+
+  /** Custom props including callbacks. */
+  overrideProps: TriggerProps;
+};
```

Comment:
I am not sure that I like the interface there, so opened for ideas/suggestions.

Reply:
IMO the interface is fine, but there should be extra documentation that deals with the priority of the prop overriding children > overrideProps > outerProps